src/test/java/com/epam/training/student_anton_lapushenko/WebDriver/Task1/BaseChromeTest.java

```diff
@@ -0,0 +1,12 @@
+package com.epam.training.student_anton_lapushenko.WebDriver.Task1;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+class BaseChromeTest extends BaseTest {
```

Comment:
I created 2 bases, one for all tests, one for chrome browser. I tried realise abstraction from OOP. I think this structure is good, because in future I would create many bases for many browsers/drivers. Is that good practice or not?

Reply:
That's not a good approach. Let's imagine you have 4000 tests that you need to run in parallel on 3 browsers. With your approach you need to create separate tests for all browsers. And you will have a lot of duplicated tests which is not OOD.
There is a pattern, WebDriver Factory, that is generally used for this purpose. You can find one of the implementation here: https://dev.to/eliasnogueira/the-best-way-to-create-browser-instances-using-the-factory-pattern-java-and-selenium-webdriver-16l5

But you can create your own implementation, but the general idea is as described in doc.

We can have different base for different test types or for tests that need different preconditions.
But it's not a good practice to use different Base for different browser types.